### PR TITLE
add golangci-lint action

### PR DESCRIPTION
**Pull request by daroga0002**
add automatic Golang CI lint action

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
+      - main
```
[the remark this answers is not in the document]
Comment (daroga0002) [reply .github/workflows/golangci-lint.yml:6]: on push event only those 2 branches will be checked (direct push to master or via PR). Any other branch checks will be done basing on pull_request events:

```
Note: By default, a workflow only runs when a pull_request's activity type is opened, synchronize, or reopened. To trigger workflows for more activity types, use the types keyword.
```

If you want I can add also any branch to check but then this workflow will be running twice. One time on pull request event on `refs/pull/:prNumber/merge` branch and second time on `GITHUB_REF` of branch

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: golangci-lint
```
[the remark this answers is not in the document]
Comment (daroga0002) [reply .github/workflows/golangci-lint.yml]: updated

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+on:
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+jobs:
+  golangci:
+    name: lint
```
[the remark this answers is not in the document]
Comment (daroga0002) [reply .github/workflows/golangci-lint.yml]: updated

File: .github/workflows/golangci-lint.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-zajac) [reply .github/workflows/golangci-lint.yml:6]: Ok, that makes sense.

**Review by m-zajac — APPROVED** [review]

**Pull request merged**